### DISCUSSION TITLE
Note about v1.22 install needing flash_nuke.uf2

### DISCUSSION THREAD

**Discussion by mendenm** [micropython/micropython#13290] [RP2040 / Pico] [answered]
I could not get an upgrade from one of the 1.22 previews to the new 1.22 release to work without running flash_nuke.uf2 to erase everything.  This is not terribly surprising, due to the size change, but if you have trouble with the upgrade, this is just a hint.

**Comment by mendenm** [comment, marked as answer]
answer is in question.

**Comment by robert-hh** [comment]
No need to nuke the device with v1.22.0 here, at least not for a plain RP2.

**Reply by mendenm** [reply]
yes, I wasn't implying a guaranteed need. I suspect it has to do with either getting unlucky with the LittleFS2 version update or the size change, so it may  be quite dependent on the previous flash contents.  I cannot, of course, test this hypothesis, since I have it thoroughly nuked.  I did have to re-do the nuke at least one more time to make the further version switches back and forth work.  Note that  I am switching from one of the v1.22 previews to the release, so it may have to do with picky details.  This on a plain Pico. YMMV.

**Reply by robert-hh** [reply]
The reason is probably not the code size. At the plain RP2, the space reserved for code is 640k, and the actual code size is about 320k.
